lib/src/api.mli

```diff
@@ -55,17 +53,17 @@ module type BASIC_CHARACTER = sig
 
   val size: t -> int
   (** Get the size of the character, the exact semantics are
-      implementation-specific (c.f. {!write_to_native_string}) *)
+      implementation-specific (c.f. {!write_to_native_bytes}) *)
 
-  val write_to_native_string: t -> buf:String.t -> index:int -> (int, [> `out_of_bounds]) result
-  (** [write_to_native_string c ~buf ~index] serializes
+  val write_to_native_bytes: t -> buf:Bytes.t -> index:int -> (int, [> `out_of_bounds]) result
+  (** [write_to_native_bytes c ~buf ~index] serializes
       the character [c] at position [index] in the native string
       [buf] (writing [size c] units). Note, as with {!size} that the
       meaning of [index] is implementation dependent (can be the {i
       index-th} byte, the {i index-th} bit, etc.). *)
 
-  val to_native_string: t -> String.t
-  (** [to_native_string c] creates a string containing the
+  val to_native_bytes: t -> Bytes.t
+  (** [to_native_bytes c] creates a string containing the
```

Comment:
This is an issue that I am undecided about, and could go either way. There seems to me an imbalance with the other functions in the API. Checkout this [branch](https://github.com/hammerlab/sosa/tree/upgrade_4023_to_native_string) for what changes. In particular, notice how the `to_native_string` method [here](https://github.com/hammerlab/sosa/blob/upgrade_4023_to_native_string/lib/src/int_utf8_character.ml#L71) has to create a bytes anyway. So why hide it? 

It's easy to change this back so if you prefer strongly let me know and we'll go with that.